backend/app/adapter/db/errors.go

```diff
@@ -0,0 +1,10 @@
+package db
+
+import "errors"
+
+var errRecordNotFound = errors.New("user does not exist")
+
+// IsErrRecordNotFound checks whether the error is ErrRecordNotFound.
+func IsErrRecordNotFound(err error) bool {
+	return errors.Is(err, errRecordNotFound)
+}
```

Comment:
Let's do something like this: https://github.com/short-d/short/blob/master/backend/app/usecase/url/urlcreator.go#L13